api/src/wrangler.rs

```diff
@@ -14,14 +14,14 @@ use fvm_shared::econ::TokenAmount;
 use fvm_shared::message::Message;
 use fvm_shared::{ActorID, MethodNum, BLOCK_GAS_LIMIT};
 use vm_api::trace::InvocationTrace;
-use vm_api::{vm_err, ActorState, MessageResult, Primitives, VMError, VM};
+use vm_api::{vm_err, ActorState, MessageResult, MockPrimitives, Primitives, VMError, VM};
 
 pub use crate::{bench::Bench, trace::ExecutionTrace, ExecutionResult};
 
 pub struct ExecutionWrangler {
     bench: RefCell<Box<dyn Bench>>,
     store: Box<dyn Blockstore>,
-    primitives: Box<dyn Primitives>,
+    primitives: Box<dyn MockPrimitives>,
```

Comment:
Without the extra MockPrimitives trait, this would have to hold a concrete FakePrimitives. 

Either design "works" in that FakePrimitives does offer complete customization over its behaviour. Though if we do not wish to couple to that particular code (imported from builtin actors) we can provide a new default here (i.e. directly in the vm crate)

Previously the vm crate duplicated code from the builtin-actors repo to provide a default implementation of FakePrimitives: Primitives.

Now we would require the vm crate to implement a FakePrimitives: MockPrimitives instead.